src/mlpack/methods/ann/layer/linear.hpp

```diff
@@ -52,6 +52,12 @@ class Linear
          const size_t outSize,
          RegularizerType regularizer = RegularizerType());
 
+  //! Copy constructor.
+  Linear(const Linear& layer);
+
+  //! Copy assignment operator.
+  Linear& operator = (const Linear& layer);
+
```

Comment:
I know you only meant to add the copy constructor in this PR, but we should add the move constructor too.  If you want, we can open an issue for it, but if you want to handle it here that works too.